internal/transport/http2_client.go

```diff
@@ -991,7 +1003,10 @@ func (t *http2Client) Close(err error) {
 		t.kpDormancyCond.Signal()
 	}
 	t.mu.Unlock()
-	t.controlBuf.finish()
+	// Per HTTP/2 spec, a GOAWAY frame must be sent before closing the
+	// connection. See https://httpwg.org/specs/rfc7540.html#GOAWAY.
+	t.controlBuf.put(&goAway{code: http2.ErrCodeNo, debugData: []byte(fmt.Sprintf("client shutdown with: %v", err)), closeConn: err})
```

Comment:
Let's omit the client shutdown error from the data we send to the server.  It doesn't seem like a good idea to send that.  It's probably just `ErrConnClosing`, but better safe than sorry.

Reply:
This still includes the error.  I would prefer if there was no dynamic data in here, just a static string that says the client is shutting down and closing the connection.

Reply:
I think this is already fixed, but i am not sure why github doesn't shows the change made post the comment.

Reply:
I don't know, but please fix it.